packages/optimizer/lib/transformers/OptimizeHeroImages.js

```diff
@@ -367,6 +370,25 @@ class OptimizeHeroImage {
     }
     appendChild(node, imgNode);
   }
+
+  isMarkedAsHeroImage(node) {
+    while (node) {
+      if (!node.tagName) {
```

Comment:
Why would `tagName` ever be empty?

Reply:
If it's not an element, but a different type of node. I initially tried to check for `HTMLElement`, but that didn't work as I expected.

Reply:
It seems elements are only ever passed into this function, but I guess this is good for hardening.

Reply:
Yes, but this check is within a loop that loops from parent to parent, so this does not only check what is passed into the function, but every ancestor of it as well. And as elements can also be children of non-elements (see https://www.w3schools.com/jsref/prop_node_nodetype.asp), I think this is needed for some edge cases.